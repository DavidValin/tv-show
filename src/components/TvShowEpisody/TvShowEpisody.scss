@import '../../shared.scss';

.TvShowEpisody {
  display: flex;
  @include fadeIn();

  @media (min-width: $screen-md-min) {
    padding: 2em;
  }
  @media (max-width: $screen-sm-max) {
    flex-wrap: wrap;
  }

  &__cover {
    img {
      width: 100%;
    }
  }

  @media (min-width: $screen-md-min) {
    &__cover,
    &__details {
      width: 50%;
    }
  }
  @media (max-width: $screen-sm-max) {
    &__cover {
      width: calc(100% - 2em);
    }
    &__details {
      width: 100%;
    }
  }

  &__cover,
  &__details {
    margin: 1em;
  }

  &__go-back {
    display: inline-block;
    margin: 1em 0;
    background: #333;
    color: white;
    padding: 0.5em 0.8em;
  }
}
