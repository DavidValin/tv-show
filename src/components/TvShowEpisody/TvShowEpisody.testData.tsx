import {IParams} from './TvShowEpisody';

export const tvShowEpisodyTestData:IParams = {
  episody: {
    id: '691591',
    title: 'Painbow',
    summary: 'A rainbow causes Townsville to become uncomfortably happy.',
    coverImg: {
      medium: 'http://static.tvmaze.com/uploads/images/medium_landscape/53/132628.jpg',
      original: 'http://static.tvmaze.com/uploads/images/original_untouched/53/132628.jpg'
    }
  }
};