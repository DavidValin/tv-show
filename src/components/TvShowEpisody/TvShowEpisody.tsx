import React from 'react';
import './TvShowEpisody.scss';
import { ITvShowEpisody } from '../../state/initial_state';
import { Link } from 'react-router-dom';
import getAvailableImage from '../../util';

export interface IParams {
  episody: ITvShowEpisody
}

export default function TvShowEpisody(props:IParams) {
  return (
    <div className='TvShowEpisody'>
      <img className='TvShowEpisody__cover' src={getAvailableImage(props.episody.coverImg)} alt={props.episody.title} />
      <div className='TvShowEpisody__details'>
        <h2 className='TvShowEpisody__details__name'>
          {props.episody.title}
        </h2>
        <p className='TvShowEpisody__details__summary'>
          {props.episody.summary}
        </p>
        <Link to={'/'} className='TvShowEpisody__go-back'>
          go back
        </Link>
      </div>
    </div>
  );
}
