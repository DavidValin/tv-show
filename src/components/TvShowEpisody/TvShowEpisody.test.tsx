import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import TvShowEpisody from './TvShowEpisody';
import { tvShowEpisodyTestData } from './TvShowEpisody.testData';

it('should match the snapshot', () => {
  const tree = renderer.create(<Router><TvShowEpisody episody={tvShowEpisodyTestData.episody} /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});