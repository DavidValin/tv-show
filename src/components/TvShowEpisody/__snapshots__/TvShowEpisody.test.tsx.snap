// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="TvShowEpisody"
>
  <img
    alt="Painbow"
    className="TvShowEpisody__cover"
    src="http://static.tvmaze.com/uploads/images/original_untouched/53/132628.jpg"
  />
  <div
    className="TvShowEpisody__details"
  >
    <h2
      className="TvShowEpisody__details__name"
    >
      Painbow
    </h2>
    <p
      className="TvShowEpisody__details__summary"
    >
      A rainbow causes Townsville to become uncomfortably happy.
    </p>
    <a
      className="TvShowEpisody__go-back"
      href="/"
      onClick={[Function]}
    >
      go back
    </a>
  </div>
</div>
`;
